src/functions-and-arrays.js

```diff
@@ -1,41 +1,142 @@
 
-function findLongestWord() {}
-
-
+function findLongestWord(arr) {
+  let logestWord = "";
+  if(!arr.length){
+    return null;
+  }else if(arr.length === 1){
+    return arr[0];
+  } else {
+    for(let i = 0; i < arr.length; i++){
+      if(logestWord.length < arr[i].length){
+        logestWord = arr[i];
+      }
+    }
+  }
+  return logestWord;
+}
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(numbers) {
+  let result = 0;
 
+  if(numbers.length === 0){
+    return 0;
+  }
 
+  for(let i = 0; i <= numbers.length - 1; i++){
+    result += numbers[i];
+  }
+  
+  return result;
+}
 
-// Iteration #3.1 Bonus:
-function sum() {}
 
 
+// Iteration #3.1 Bonus:
+const mixedArr = [6, 12, 'miami', 1, true, 'barca', '200', 'lisboa', 8, 10];
+
+function sum(mixedArr) {
+  let sumResult = 0;
+  console.log(typeof mixedArr);
+
+  if(mixedArr.length === 0){
+    return sumResult;
+  }
+
+  for(let i = 0; i < mixedArr.length; i++){
+    if(typeof mixedArr[i] === "string"){
+      sumResult += mixedArr[i].length; 
+    }else if(typeof mixedArr[i] === "number"){
+      sumResult += mixedArr[i];
+    }else if( typeof mixedArr[i] === "boolean"){
+      if(mixedArr[i] === true){
+        sumResult += 1;
+      }
+    }else if(typeof mixedArr[i] !== "boolean" && typeof mixedArr[i] !== "string" && typeof mixedArr[i] !== "number" ){
+      throw new Error("error unsupported type");
+    }
+  }
+  return sumResult;
+}
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(sumNumbersAvg) {
+  let sumThings = 0;
+
+  if(sumNumbersAvg.length === 0){
+    return null;
+  }
+
+  sumNumbersAvg.forEach((sumNumberAvg) => {
+    sumThings += sumNumberAvg;
+  });
+
+  return sumThings / sumNumbersAvg.length;
+}
 
 
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
-function averageWordLength() { }
+function averageWordLength(avgWordArr) {
+  let wordsLength = 0;
+
+  if(avgWordArr.length === 0){
+    return null;
+  }
+
+  for(let i = 0; i < avgWordArr.length; i++){
+    wordsLength += avgWordArr[i].length;
+  }
+
+  return wordsLength / avgWordArr.length;
+}
 
 // Bonus - Iteration #4.1
-function avg() {}
+function avg(mixedArr) {
+  let sumResult = 0;
+
+  if(mixedArr.length === 0){
+    return null;
+  }
+
+  for(let i = 0; i < mixedArr.length; i++){
+    if(typeof mixedArr[i] === "string"){
+      sumResult += mixedArr[i].length; 
+    }else if(typeof mixedArr[i] === "number"){
+      sumResult += mixedArr[i];
+    }else if( typeof mixedArr[i] === "boolean"){
+      if(mixedArr[i] === true){
+        sumResult += 1;
+      }
+    }else{
+      throw new Error("error unsupported type");
+    }
```

Comment:
since you already did the `function sum()` you can reuse it here:

```
function avg (mixedArr) {
  if (mixedArr.length === 0) return null;
  else return sum(mixedArr) / mixedArr.length;
}
```